### PR TITLE
#502 security module clair integration

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description
To make CICD light we are breaking Security module (Clair) separately and listing it on the modules pages. So if user wants it he can install it.

Divide CICD into 2 part
CICD and Security module (Clair)

Fixes # [(issue)](https://github.com/devtron-labs/dashboard/issues/502)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- verified security option should not be there in primary navigation if Security module is not installed
- Go to Deploy image section and click on Show source info the Security tab should only be visible  if Security module is installed
- Go to CI Pipeline configuration the Scan for vulnerabilities box on the build page should only be visible  if Security module is installed
- Go to Build history page the Security tab should only be visible  if Security module is installed

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/CIPipelineN/CIPipeline.tsx [hunk 1]
```diff
@@ -74,6 +76,7 @@ export default function CIPipeline({
     }>({ preBuildStage: [], postBuildStage: [] })
     const [presetPlugins, setPresetPlugins] = useState<PluginDetailType[]>([])
     const [sharedPlugins, setSharedPlugins] = useState<PluginDetailType[]>([])
+    const [isSecurityModuleInstalled, setIsSecurityModuleInstalled] = useState<boolean>(false)
```
Comment (sohel-rp) [inline src/components/CIPipelineN/CIPipeline.tsx:79]: Change `setIsSecurityModuleInstalled` to `setSecurityModuleInstalled`

**Review by sohel-rp — APPROVED** [review]

File: src/components/CIPipelineN/Build.tsx [hunk 2]
```diff
@@ -18,11 +18,13 @@ export function Build({
     isAdvanced,
     ciPipelineId,
     pageState,
+    isSecurityModuleInstalled
 }: {
```
Comment (shivani170) [inline src/components/CIPipelineN/Build.tsx:22]: Why arent we making interface object for type

**Review by shivani170 — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]

File: src/components/v2/devtronStackManager/DevtronStackManager.tsx [hunk 3]
```diff
@@ -219,10 +217,7 @@ export default function DevtronStackManager({
 
     const _getGitOpsConfigurationStatus = async (): Promise<void> => {
         try {
-            if (
-                location.pathname.includes('/details') &&
-                queryParams.get('id') === ModuleNameMap.ARGO_CD
-            ) {
+            if (location.pathname.includes('/details') && queryParams.get('id') === ModuleNameMap.ARGO_CD) {
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.tsx:220]: Can we use `details` from constant

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 4]
```diff
@@ -90,36 +89,46 @@ const ModuleDetailsCard = ({
...
+          {getInstallationStatusLabel(moduleDetails.installationStatus)}
+          <img className="module-details__card-icon mb-16" src={moduleDetails.icon} alt={moduleDetails.title} />
+          <div className="module-details__card-name fs-16 fw-6 cn-9 mb-4">{moduleDetails.title}</div>
+          <div className="module-details__card-info fs-13 fw-4 cn-7 lh-20">
+              {moduleDetails.name === MORE_MODULE_DETAILS.name ? (
+                  <>
+                      You can&nbsp;
+                      <a
+                          href="https://github.com/devtron-labs/devtron/issues/new/choose"
+                          className="cb-5 fw-6"
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:118]: On hover,  it must be changing the text color. Is it okay to have that please do confirm utkarsh

File: src/css/base.scss [hunk 5]
```diff
@@ -1735,6 +1735,10 @@ button.anchor {
     border-bottom: solid 1px var(--N100);
 }
 
+.dc__border-bottom-n2 {
+    border-bottom: solid 1px var(--N200);
```
Comment (sohel-rp) [inline src/css/base.scss:1739]: Please use existing `border-bottom` class.

Comment (vivek-devtron) [reply src/css/base.scss:1739]: that is getting overwritten by border-bottom from nodeType.scss

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:118]: it is already handled

File: src/components/v2/devtronStackManager/DevtronStackManager.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.tsx:220]: done

File: src/config/constants.ts [hunk 6]
```diff
@@ -187,7 +187,7 @@ export const AppConfigStatus = {
 
 export const FullRoutes = {
     LOGIN: `${Routes.LOGIN}`,
-    CENTRAL: 'https://api.devtron.ai',
+    CENTRAL: 'https://api-stage.devtron.ai',
```
Comment (sohel-rp) [inline src/config/constants.ts:190]: Please change to prod url

Comment (vivek-devtron) [reply src/config/constants.ts:190]: Will do before merge

File: src/components/app/details/triggerView/cdMaterial.tsx [hunk 7]
```diff
@@ -12,8 +12,32 @@ import { EmptyStateCdMaterial } from './EmptyStateCdMaterial';
...
+    this.getSecurityModuleStatus()
+  }
+
+  async getSecurityModuleStatus(): Promise<void> {
+    try {
+        const { result } = await getModuleInfo(ModuleNameMap.SECURITY)
+        if (result?.status === ModuleStatus.INSTALLED) {
+            this.setState((prevState) => ({
+                isSecurityModuleInstalled: true,
+            }))
```
Comment (sohel-rp) [inline src/components/app/details/triggerView/cdMaterial.tsx:37]: Can be changed to `this.setState({ isSecurityModuleInstalled: true })`?

File: src/components/app/details/triggerView/cdMaterial.tsx [hunk 8]
```diff
@@ -135,28 +159,30 @@ export class CDMaterial extends Component<CDMaterialProps> {
               : mat.isSelected ? <Check className="align-right icon-dim-24" /> : "Select"}
           </div>
         </div>
+        {!this.props.isFromDeploymentGroup && <>
         {mat.showSourceInfo ? <>
```
Comment (sohel-rp) [inline src/components/app/details/triggerView/cdMaterial.tsx:163]: Can we change this to `&&` pattern?

File: src/components/app/details/triggerView/cdMaterial.tsx (hunk 7 above)
Comment (vivek-devtron) [reply src/components/app/details/triggerView/cdMaterial.tsx]: done

File: src/components/app/details/triggerView/cdMaterial.tsx (hunk 8 above)
Comment (vivek-devtron) [reply src/components/app/details/triggerView/cdMaterial.tsx]: done

**Review by sohel-rp — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]

**Pull request merged**